app/models/application_record.rb

```diff
@@ -21,4 +23,17 @@ def append_url_options(path, options = {})
 
     path
   end
+
+private
+
+  def normalise_locale(locale)
+    case locale
+    when Struct
+      locale.to_a[0]
```

Comment:
I think there's a `code` method on the `Struct` that can be used instead of making an assumption the code is always the first element.

Reply:
Sorted 👍